### PR TITLE
Tweak: Additional Breakpoints - System tweaks (ED-3907, ED-3852, ED-3999, ED-3943, ED-4123)

### PR DESCRIPTION
**Pull request by udidol**
* Tweak: Removed Redundant `add_feature` call from a Breakpoints Manager Unit test.
* Tweak: Moved all usages of `Controls_Stack` responsive constants to Breakpoints/Manager constants.

### CODE REVIEW

**Comment by rodolphebertozzo** [comment]
Please, see : https://github.com/elementor/elementor/issues/15473

File: assets/dev/scss/frontend/_devices.scss [hunk 1]
```diff
@@ -2,16 +2,32 @@
...
 		@media (min-width: $screen-lg-min) {
 			content: 'desktop';
 		}
 
+		@media (max-width: $screen-laptop-max) {
+			content: 'laptop';
+		}
+
+		@media (max-width: $screen-tablet-extra-max) {
+			content: 'tablet-extra';
```
Comment (kobizz) [inline assets/dev/scss/frontend/_devices.scss:18]:
```suggestion
			content: 'tablet_extra';
```

File: assets/dev/scss/frontend/_devices.scss [hunk 2]
```diff
@@ -2,16 +2,32 @@
...
+		}
+
 		@media (min-width: $screen-md-min) and (max-width: $screen-md-max) {
 			content: 'tablet';
 		}
 
+		@media (max-width: $screen-mobile-extra-max) {
+			content: 'mobile-extra';
```
Comment (kobizz) [inline assets/dev/scss/frontend/_devices.scss:26]:
```suggestion
			content: 'mobile_extra';
```

File: core/breakpoints/manager.php [hunk 3]
```diff
@@ -110,15 +111,23 @@ public function get_active_breakpoints( $breakpoint_name = null ) {
...
 	 *
 	 * @since 3.2.0
 	 *
 	 * @return boolean
 	 */
 	public function has_custom_breakpoints() {
-		$breakpoints = $this->get_breakpoints();
+		$breakpoints = $this->get_active_breakpoints();
+
+		$additional_breakpoints = [ 'mobile_extra', 'tablet_extra', 'laptop', 'widescreen' ];
```
Comment (kobizz) [inline core/breakpoints/manager.php:124]: Use constants

File: assets/dev/scss/helpers/variables.scss [hunk 4]
```diff
@@ -220,9 +220,13 @@ $panel-xl-width: 300px;
...
 		sm: 480px, // Mobile landscape
 		md: 768px, // Tablet
+		mobile-extra: 880px,
 		lg: 1025px, // Desktop
+		tablet-extra: 1200px,
+		laptop: 1366,
 		xl: 1440px, // Larger devices
-		xxl: 1600px // Largest devices
+		xxl: 1600px, // Largest devices
+		widescreen: 2400px
```
Comment (kobizz) [inline assets/dev/scss/helpers/variables.scss:229]: Check if we need these new variables

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/responsive/files/frontend.php [hunk 5]
```diff
@@ -54,9 +63,9 @@ public function parse_content() {
 					$is_max_point = 'MAX' === $placeholder_data[2];
 
 					// If the placeholder capture is `MOBILE_NEXT` or `TABLET_NEXT`, the original breakpoint value is used.
-					if ( ! $is_max_point && 'NEXT' !== $placeholder_data[2] ) {
-						$breakpoint_index--;
-					}
+				if ( ! $is_max_point && 'NEXT' !== $placeholder_data[2] ) {
+					$breakpoint_index--;
+				}
```
Comment (kobizz) [inline core/responsive/files/frontend.php:68]: Fix PHP-Lint

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/scss/frontend/_devices.scss (hunk 1 above)
Comment (udidol) [reply assets/dev/scss/frontend/_devices.scss]: Done.

File: assets/dev/scss/frontend/_devices.scss (hunk 2 above)
Comment (udidol) [reply assets/dev/scss/frontend/_devices.scss]: Done.

File: core/breakpoints/manager.php (hunk 3 above)
Comment (udidol) [reply core/breakpoints/manager.php]: Done.

File: assets/dev/scss/helpers/variables.scss (hunk 4 above)
Comment (udidol) [reply assets/dev/scss/helpers/variables.scss]: Seems we don't. Removed.

File: core/responsive/files/frontend.php (hunk 5 above)
Comment (udidol) [reply core/responsive/files/frontend.php]: Done.

**Pull request merged**